crates/ruma-common/src/api.rs

```diff
@@ -526,11 +526,17 @@ pub enum AuthScheme {
     /// Using the query parameter is deprecated since Matrix 1.11.
     AccessTokenOptional,
 
+    /// Authentication is performed by including an appservice access token in the `Authentication`
```

Comment:
It believe that we need to clarify the difference between both variants. Like `AppserviceToken` are endpoints only accessible by appservices and they require a token, and `AppserviceTokenOptional` are endpoints accessible by both appservices and clients, and the appservices can provide a token.

Reply:
I like the first one, but I find the second one hard to parse. Maybe this:

```
    /// Authentication is not required for clients, but should be provided for appservices, by including an
    /// appservice access token in the `Authentication` http header, or an `access_token` query parameter.
```